src/components/AssetName/AssetName.tsx

```diff
@@ -0,0 +1,72 @@
+import type { TextProps } from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import { useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+
+import { RawText } from '@/components/Text'
+import type { TranslateFunction } from '@/lib/translate'
+import { chainIdToChainDisplayName } from '@/lib/utils'
+import { selectAssets } from '@/state/slices/selectors'
+import { useAppSelector } from '@/state/store'
+
+type AssetNameProps = {
+  assetId: AssetId
+  postFix?: string
+  showAssetSymbol?: boolean
+  ref?: React.RefObject<HTMLDivElement> | ((el: HTMLElement | null) => void)
+} & TextProps
+
+export const buildAssetTitle = (
+  asset: Asset,
+  translate: TranslateFunction,
+  postFix?: string,
+  showAssetSymbol?: boolean,
+): string => {
+  if (showAssetSymbol && postFix) {
+    return `${asset.symbol} ${postFix}`
+  }
+
+  if (showAssetSymbol) {
+    return asset.symbol
+  }
+
+  if (postFix) {
+    return `${asset.name} ${postFix}`
+  }
+
+  if (asset.relatedAssetKey && asset.relatedAssetKey !== asset.assetId) {
+    return translate('common.assetOnChain', {
+      assetName: asset.name,
+      chainName: chainIdToChainDisplayName(asset.chainId),
+    })
+  }
+
+  return asset.name
+}
+
+export const AssetName = ({
```

Comment:
Same here - do we need an explicit component just to display this? i.e, if we have it in the asset in store, then that diff could also be reverted

Reply:
Nop, removed it to add at runtime